### PR TITLE
Browseraction panel 1.6

### PR DESCRIPTION
**Pull request by groovecoder**

### CODE REVIEW

File: src/fbc-icon.svg [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="28px" height="24px" viewBox="0 0 28 24" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <!-- Generator: Sketch 53.2 (72643) - https://sketchapp.com -->
```
Comment (jonathanKingston) [inline src/fbc-icon.svg]: Remove this please and the other sketch related tags/comments. All of the id's can be removed too.

File: src/panel.css [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  }
+
+  .text{
+      padding-left: 25px;
+      padding-right: 45px;
+      font-size: 14px;
+  }
+
+  .img {
+      padding-right: 10px;
```
Comment (jonathanKingston) [inline src/panel.css]: There are 2 and 4 space indenting in this file.

File: src/background.js [hunk 3]
```diff
@@ -322,7 +322,29 @@ function stripFbclid(url) {
...
+
+async function updateBrowserActionIcon (url) {
+  if(isFacebookURL(url)) {
+    browser.browserAction.setPopup({popup: "./panel1.html"});
+    const FBC_STORAGE = await browser.storage.local.get();
+    if (FBC_STORAGE.PANEL_SHOWN !== true) {
+      await browser.browserAction.setBadgeBackgroundColor({
+        color: "#3B5998"
+      });
+      browser.browserAction.setBadgeText({text: "funny"});
```
Comment (jonathanKingston) [inline src/background.js]: Change the text to a real translation string.

File: src/background.js [hunk 4]
```diff
@@ -322,7 +322,29 @@ function stripFbclid(url) {
...
+    }
+  } else {
+    browser.browserAction.setPopup({popup: "./panel2.html"});
+    browser.browserAction.setBadgeText({text: ""});
+  }
+}
+
 async function containFacebook (options) {
+  await updateBrowserActionIcon (options.url);
```
Comment (jonathanKingston) [inline src/background.js]: The space after the function name declaration I find a weird but it's used elsewhere. However the space on call sites appears not to be.
The current style appears to be:
```
function name () {
}
name();
```

File: src/background.js [hunk 5]
```diff
@@ -322,7 +322,29 @@ function stripFbclid(url) {
   return strippedUrl.href;
 }
 
+async function tabUpdateListener (tabId, changeInfo, tab) {
+  await updateBrowserActionIcon(tab.url);
+}
+
+async function updateBrowserActionIcon (url) {
+  if(isFacebookURL(url)) {
```
Comment (jonathanKingston) [inline src/background.js]: Space after if.

File: src/background.js [hunk 6]
```diff
@@ -322,7 +322,29 @@ function stripFbclid(url) {
...
+    browser.browserAction.setPopup({popup: "./panel1.html"});
+    const FBC_STORAGE = await browser.storage.local.get();
```
Comment (jonathanKingston) [inline src/background.js]: Global const's are generally uppercased where things are 100% constant. So like magic variables like `const ICON_COLOR = "red";` but `const fbcStorage = await ...` is more about preventing reassignment of  a dynamic variable.

Elsewhere is using `const url = new URL(options.url);` for example.

File: src/background.js (hunk 3 above)
Comment (ShivangiKakkar) [reply src/background.js]: I wasn't sure which string to add here replacement of this. So except this, all the requested changes are done. :)

File: src/panel2.html [hunk 7]
```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE html>
+<html id="2">
+  <head>
+    <meta http-equiv="content-type" content="text/html; charset=utf-8">
+    <title>Facebook Container</title>
+    <link rel="stylesheet" href="panel.css">
+    <script src="background.js"></script>
```
Comment (stoically) [inline src/panel2.html]: The background script should not be included in the popup.

File: src/background.js (hunk 3 above)
Comment (groovecoder) [reply src/background.js]: Oops, yeah I was trying different text strings to see how much of each would show up in the badge.

**Comment by groovecoder** [comment]
Okay, I've added the i18n plumbing to:

* Show the translated add-on description in about:addons
* Show the translated panel UI strings

Testing this is a bit complicated, because there seems to be [a bug in Firefox](https://bugzilla.mozilla.org/show_bug.cgi?id=1524679), [a bug in web-ext](https://github.com/mozilla/web-ext/issues/1498), or both. So, to test this ...

1. Create a new Firefox profile
2. Open Firefox with the new profile, and under languages, click your current language and then select "Search for more langauges"
image 
3. Click "Select languages to add" and then choose "German" and then click "Add" (Note: use German because that's the language I Google-translated for testing)
image
  * Firefox will download and install the German language pack. Make sure it's the first/top language and restart Firefox with this profile.
4. When Firefox restarts, you should see it open with all-German UI. Go to `about:debugging` and click "Add-on temporar laden...", and choose the `manifest.json` file from this branch's `src/` directory.
5. Go to `about:addons` and you should see the add-on description in German ...
image
6. Go to www.facebook.com, click the `browserAction` icon to open the pop-up, and you should see the Facebook panel content in German ...
image
7. Go to a non-Facebook site, again click the `browserAction` icon to open the pop-up, and you should see the non-Facebook panel content in German ...
image
8. Rejoice - the add-on UI is now internationalized and localized!

@flodolo - can you take a look at this? Does this look like the right way to structure these `messages` files for adding to pontoon?

File: src/_locales/en/messages.json [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    "extensionName": {
+        "message": "Facebook Container",
+        "description": "Name of the extension."
+    },
+    "extensionDescription": {
+        "message": "Facebook Container isolates your Facebook activity from the rest of your web activity in order to prevent Facebook from tracking you outside of the Facebook website via third party cookies.",
+        "description": "Description of the extension."
+    },
+    "onFacebookHeader": {
+        "message": "You're browsing in a Facebook Container tab",
```
Comment (flodolo) [inline src/_locales/en/messages.json]: We should use proper apostrophes in the strings `’`.

Comment (groovecoder) [reply src/_locales/en/messages.json]: This one? ’

Comment (flodolo) [reply src/_locales/en/messages.json]: Yes

File: src/panel.js [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+document.addEventListener("DOMContentLoaded", async () => {
+  if (document.querySelector("#panel1") !== null) {
+    await browser.storage.local.set({PANEL_SHOWN: true});
+    await browser.browserAction.setBadgeText({text: ""});
+  }
+
+  const tabsQueryResult = await browser.tabs.query({currentWindow: true, active: true});
+  const currentActiveTab = tabsQueryResult[0];
+  const currentActiveURL = new URL(currentActiveTab.url);
+  const currentActiveHostname = currentActiveURL.hostname;
```
Comment (stoically) [inline src/panel.js]: `hostname` will be empty for urls starting with `about:` and hence the header text in the popup wont start with an url.

**Comment by stoically** [comment]
To get rid of the TypeError's in the test output you could add the following in the `test/setup.js` as additional property in the second parameter of the `webExtensionsJSDOM.fromManifest` call:

```js
    popup: {
      jsdom: {
        beforeParse: (window) => {
          window.browser.tabs._create({});
        }
      }
    }
```

That's because the `panel.js` queries for an active tab on load which is not present otherwise.

Btw, another thing to maybe consider might be adding `webExtension.background.browser.i18n.getUILanguage.returns("en_US");` to the setup - which would be an alternative to checking for a return value in the background script.

**Comment by mathjazz** [comment]
> @mathjazz can you take a look
> 
> I think this makes sense to me, but it's also the first WebExtension we add to Pontoon, and I want to double check.

It looks good to me.

I've set it up on stage for testing:
https://mozilla-pontoon-staging.herokuapp.com/de/contain-facebook/messages.json/

**Comment by flodolo** [comment]
Thanks Matjaz. Side question, where is that German translation coming from?

**Comment by mathjazz** [comment]
From the repo - I used this branch:
https://github.com/mozilla/contain-facebook/tree/browseraction-panel-1.6/src/_locales

**Comment by flodolo** [comment]
> From the repo - I used this branch:
> https://github.com/mozilla/contain-facebook/tree/browseraction-panel-1.6/src/_locales

Sorry, the question was for Luke.

**Comment by groovecoder** [comment]
Yes the German in this branch is for testing. I just used Google Translate to translate the English. Should be replaced on Pontoon.

I'm superseding this PR with https://github.com/mozilla/contain-facebook/pull/214 which includes @ShivangiKakkar's commit for dark/light icons, and a fix for @stoically's comment on the URL for `about:` pages.

**Pull request closed without merging**